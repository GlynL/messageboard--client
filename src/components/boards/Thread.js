import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import * as boardActions from "../../actions/boards";
import * as threadActions from "../../actions/threads";
import isObjectEmpty from "../../helpers/isObjectEmpty";
import Replies from "./Replies";

class Thread extends Component {
  componentDidMount() {
    this.props.fetchBoards();
  }

  componentDidUpdate(prevProps, prevState) {
    // TODO: this flow not working - setup single lookup api?
    this.props.setActive(this.props.boards, this.props.match.params.board);
    if (!isObjectEmpty(this.props.board)) {
      this.props.setActiveThread(
        this.props.match.params.thread,
        this.props.board
      );
    }
  }

  render() {
    if (isObjectEmpty(this.props.thread)) {
      return <div>loading...</div>;
    }
    return (
      <div>
        <Link to={`/${this.props.board.name}`}>
          Back to {this.props.board.name}
        </Link>
        <h1>{this.props.thread.title}</h1>
        <p>{this.props.thread.text}</p>
        <Replies />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  board: state.boards.activeBoard,
  boards: state.boards.boards,
  thread: state.threads.activeThread
});

export default connect(
  mapStateToProps,
  { ...boardActions, ...threadActions }
)(Thread);
