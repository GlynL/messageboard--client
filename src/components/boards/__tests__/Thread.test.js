import React from "react";
import { shallow } from "enzyme";
import { Thread } from "../Thread";

describe("initial load", () => {
  let setActive, setActiveThread, match, wrapper;
  beforeEach(() => {
    setActive = jest.fn();
    setActiveThread = jest.fn();
    match = {
      params: {
        board: "mock board"
      }
    };
    wrapper = shallow(
      <Thread
        match={match}
        setActive={setActive}
        thread={{}}
        setActiveThread={setActiveThread}
      />
    );
  });

  it("calls componentDidMount", () => {
    expect(setActive).toHaveBeenCalledWith(match.params.board);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("w/ board prop", () => {
  let match, wrapper, board, setActiveThread;

  beforeEach(() => {
    board = {
      name: "mock board"
    };
    match = {
      params: {
        board: "mock board",
        thread: "mock thread"
      }
    };
    setActiveThread = jest.fn();
    wrapper = shallow(
      <Thread
        match={match}
        thread={{}}
        // thread={{ title: "mock thread", text: "mock text" }}
        board={board}
        setActive={jest.fn()}
        setActiveThread={jest.fn()}
      />
    );
  });

  // it("call componentDidUpdate", () => {
  //   // wrapper.setProps({ setActiveThread });
  //   expect(setActiveThread).toHaveBeenCalledWith(match.params.thread, board);
  // });

  it("matches snapshot w/ thread", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
