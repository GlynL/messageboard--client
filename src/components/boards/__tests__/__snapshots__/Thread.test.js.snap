// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial load matches snapshot 1`] = `<_default />`;

exports[`w/ board prop matches snapshot w/ thread 1`] = `<_default />`;
